plugins/@grouparoo/airtable/src/lib/destination/destinationOptions.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  DestinationOptionsMethod,
+  DestinationOptionsMethodResponse,
+  SimpleDestinationOptions,
+} from "@grouparoo/core";
+import { Table } from "../client/models";
+import { IClient } from "../client/interfaces/iClient";
+
+export interface AirtableDestinationOptions extends SimpleDestinationOptions {
+  tableId: string;
+  primaryKey: string;
+}
+
+export const destinationOptions: DestinationOptionsMethod<IClient> = async ({
+  connection,
+  destinationOptions,
+}) => {
+  const optionHandler = new DestinationOptionsHandler(connection);
+
+  return await optionHandler.getDestinationOptions(
+    destinationOptions as AirtableDestinationOptions
+  );
+};
+
+class DestinationOptionsHandler {
+  private client: IClient;
+  constructor(connection: IClient) {
+    this.client = connection;
+  }
+
+  public async getDestinationOptions(
+    destinationOptions: AirtableDestinationOptions
+  ): Promise<DestinationOptionsMethodResponse> {
+    const out: DestinationOptionsMethodResponse = {};
+    Object.assign(out, await this.getRecordOptions(destinationOptions));
+    return out;
+  }
+
+  private async getTables(): Promise<Map<string, Table>> {
+    const tables = await this.client.listTables();
+    return new Map(tables.map((table) => [table.id, table]));
+  }
+
+  private async getSchema(tableId: string): Promise<string[]> {
+    return this.getTableFields(tableId);
+  }
+
+  private async getRecordOptions(
+    destinationOptions: AirtableDestinationOptions
+  ): Promise<DestinationOptionsMethodResponse> {
+    const out: DestinationOptionsMethodResponse = {
+      tableId: { type: "list", options: [], descriptions: [] },
+      primaryKey: { type: "pending", options: [] },
+    };
+    const tables = await this.getTables();
+    const sortedTables = Array.from(tables)
+      .map(([, table]) => {
+        return {
+          option: table.id,
+          description: table.name,
+        };
+      })
+      .sort((a, b) => a.description.localeCompare(b.description));
+
+    out.tableId.options = sortedTables.map((value) => value.option);
+    out.tableId.descriptions = sortedTables.map((value) => value.description);
+
+    const tableId = destinationOptions.tableId;
+    if (tableId) {
+      const fields = await this.getSchema(tableId);
+      out.primaryKey.type = "typeahead";
+      out.primaryKey.options = fields;
```

Comment:
I see. I can add a guard there when getting possible fields that only returns fields that are single line of text, dates, and emails

Reply:
Many APIs have a `calculated: true` kind of idea, but I don't see that. Just types.
Here's the list: https://airtable.com/api/meta
Ones that definitely look calculated:
* formula
* lastModifiedBy
* lastModifiedTime
* rollup?
* autoNumber?

A lot of the other ones more than you mentioned look settable (rating?, checkbox?, multilineText).

We've found it's less frustrating for people to include too much instead of leaving one out, so probably have a blacklist of types to leave out.

Reply:
Oh wait! They have `Permission levels` on there.
That should do it.